src/g_text.c

```diff
@@ -231,15 +231,32 @@ void canvas_iemguis(t_glist *gl, t_symbol *guiobjname)
     t_binbuf *b = binbuf_new();
     int xpix, ypix;
 
+    // enable autoconnect
+    int connectme, indx, nobj;
+    canvas_howputnew(gl, &connectme, &xpix, &ypix, &indx, &nobj);
+
+
     pd_vmess(&gl->gl_pd, gensym("editmode"), "i", 1);
-    glist_noselect(gl);
+ //   glist_noselect(gl);
+
     SETSYMBOL(&at, guiobjname);
     binbuf_restore(b, 1, &at);
-    glist_getnextxy(gl, &xpix, &ypix);
-    canvas_objtext(gl, xpix/gl->gl_zoom, ypix/gl->gl_zoom, 0, 1, b);
-    canvas_startmotion(glist_getcanvas(gl));
-    canvas_undo_add(glist_getcanvas(gl), UNDO_CREATE, "create",
-        (void *)canvas_undo_set_create(glist_getcanvas(gl)));
+    if (connectme)
+    {
+	canvas_objtext(gl,  xpix, ypix, 0, 1, b);
+        canvas_connect(gl, indx, 0, nobj, 0);
+    }
+    else 
+    {
+	glist_getnextxy(gl, &xpix, &ypix);
```

Comment:
that's superfluous. `canvas_howputnew()` already calls `glist_getnextxy()` (and dezooms the coordinates) if it detects that autoconnect is not desired.

so the code could be much simplified, see e.g. the implementation in `canvas_msg()`: https://github.com/pure-data/pure-data/blob/master/src/g_text.c#L490-L504